packages/espressocash_app/lib/features/ramp/screens/ramp_partner_select_screen.dart

```diff
@@ -1,83 +1,215 @@
+}
+
+class _RampPartnerSelectScreenState extends State<RampPartnerSelectScreen> {
+  Country? _country;
+
+  IList<RampPartner> get _partners => switch (widget.type) {
+        RampType.onRamp => getOnRampPartners(_country?.code),
+        RampType.offRamp => getOffRampPartners(_country?.code),
+      };
+
+  @override
+  void initState() {
+    super.initState();
+
+    final repository = sl<ProfileRepository>();
+
+    final country = repository.country;
+    if (country != null) {
+      _country = Country.findByCode(country);
+    }
+  }
+
+  void _handleCountryChange(Country country) => runWithLoader(
+        context,
+        () async {
+          await sl<UpdateProfile>()
+              .call(countryCode: country.code)
+              .foldAsync((e) => throw e, ignore);
+
+          if (!mounted) return;
+
+          setState(() {
+            _country = country;
+          });
         },
-        onContinue: () => onPartnerSelected(topPartner),
-        onMoreOptions: () => RampMoreOptionsScreen.push(
+        onError: (error) => showErrorDialog(
           context,
-          type: type,
-          otherPartners: otherPartners,
-          onPartnerSelected: (RampPartner partner) {
-            Navigator.pop(context);
-            onPartnerSelected(partner);
-          },
+          context.l10n.lblProfileUpdateFailed,
+          error,
         ),
       );
+
+  @override
+  Widget build(BuildContext context) => CpTheme.black(
+        child: Scaffold(
+          appBar: CpAppBar(
+            leading: const CpBackButton(),
+            title: Text(
+              switch (widget.type) {
+                RampType.onRamp => context.l10n.ramp_btnAddCash.toUpperCase(),
+                RampType.offRamp => context.l10n.ramp_btnCashOut.toUpperCase(),
+              },
+            ),
+          ),
+          body: OnboardingScreen(
+            children: [
+              if (_partners.isEmpty) ...[
+                Assets.icons.errorIcon.svg(height: 100.h),
+              ] else
+                switch (widget.type) {
+                  RampType.onRamp =>
+                    Assets.images.cashInGraphic.image(height: 100.h),
+                  RampType.offRamp =>
+                    Assets.images.cashOutGraphic.image(height: 100.h),
+                },
+              SizedBox(height: 20.h),
+              Padding(
+                padding: EdgeInsets.symmetric(horizontal: 42.w),
+                child: Text.rich(
+                  TextSpan(
+                    style: TextStyle(
+                      fontSize: 32.sp,
+                      fontWeight: FontWeight.w900,
+                      letterSpacing: 0.25,
+                      height: 1,
+                    ),
+                    children: <TextSpan>[
+                      TextSpan(
+                        text: switch (widget.type) {
+                          RampType.onRamp => context
+                              .l10n.selectPartnerOnRampTitle.firstWord
```

Comment:
Don't apply formatting to text based on the word position – it will break with localization. Use markdown widget with some custom style.

Reply:
updated